### DISCUSSION TITLE
Parent Grain Reconstruction

### DISCUSSION THREAD

**Discussion by Achintya-Patra** [mtex-toolbox/mtex#1564] [Ask Anything]
I am trying to do a Parent Grain Reconstruction for my material which is predominantly martensitic in nature. I tried using the code available in the documentation for the initial analysis. However, when I am trying to merge the similarly orientated grains using the mergeSimilar command, I am getting an error as given below.

_Unable to perform assignment because the left and right sides have a different number of elements._

Can anyone please suggest what I might me doing wrong here and how to resolve this issue

**Comment by zmichels** [comment]
Probably folks will need to see more of your code and progressive output to help diagnose.

Zach

**Reply by Achintya-Patra** [reply]
Since I am trying this for the first time, I have used the same code present in the documentation for initial analysis.
https://mtex-toolbox.github.io/GrainGraphBasedReconstruction.html

I am able to get the parent orientations of the individual grains. However, when I am trying to merge grains with similar orientations, it is showing an error. The microstructure after computing the parent orientation of each child grain is as follows.

![D2](https://user-images.githubusercontent.com/73560442/190863828-a139b7e3-d9d4-4a02-8c11-4a3422f600e2.png)


The error message is as follows

Unable to perform assignment because the left and right sides have a different number of elements.
Error in grain2d/merge (line 272)
      grainsMerged.phaseId(i) = find(newPhase);
Error in parentGrainReconstructor/mergeSimilar (line 19)
[job.grains, mergeId] = merge(job.grains, 'threshold',5*degree, varargin{:});

**Reply by Tijmenvermeij** [reply]
Hi, 
I've seen this error before. I think it was because I had duplicate crystal symmetries in the ebsd/grain variable, or crystal symmetries that were almost exactly the same.

**Reply by Achintya-Patra** [reply]
@Tijmenvermeij could you please elaborate, also how did you resolve this issue?

**Reply by Tijmenvermeij** [reply]
You should check the crystal symmetries in your ebsd/grain variable, by e.g. checking each crystalSymmetry (CS) in ebsd.CSList ebsd.CSList{2} and seeing if any of them are (close to) the same. If you want, I can check it if you send the data (e.g. the job variable).
I fixed it by removing the duplicate CS, which was there because I erroneously added the parent phase two times.

**Comment by zmichels** [comment]
Perhaps you can try running the example with the example dataset to make sure that works for you. If that works, then the error is likely related to whatever you did to adapt the example for your dataset. And if that’s the case… you may need to share more details about what you did whike attempting to following the example but with different data.

**Reply by Achintya-Patra** [reply]
Initially I tried using the example dataset to understand the process and it worked fine. This issue only happens when I am using my dataset. The dataset I am using has both fcc and bcc phases, although it is predominantly bcc with less than 1% points indexed as fcc. To carry out reconstruction process I have only partitioned out the bcc part. Other than this change in the dataset, rest of the code is same as the shared code in the documentation.

**Comment by ralfHielscher** [comment]
If nothing else helps, you can privately send me the data and your script and I will take a look on it.

Ralf.
